defold-spine/src/comp_spine_model.cpp

```diff
@@ -727,25 +734,32 @@ namespace dmSpine
             }
 
             component.m_DoRender = 1;
+
+            // Update bounding boxes
+            SpineModelBounds& bounds = world->m_BoundingBoxes[i];
+            GetSkeletonBounds(component.m_SkeletonInstance,bounds);
         }
 
         // Since we've moved the child game objects (bones), we need to sync back the transforms
         update_result.m_TransformsUpdated = num_active > 0;
         return dmGameObject::UPDATE_RESULT_OK;
     }
 
+
     static void RenderBatch(SpineModelWorld* world, dmRender::HRenderContext render_context, dmRender::RenderListEntry *buf, uint32_t* begin, uint32_t* end)
     {
         //DM_PROFILE(SpineModel, "RenderBatch");
 
-        const SpineModelComponent* first = (SpineModelComponent*) buf[*begin].m_UserData;
+        uint32_t component_index = (uint32_t)buf[*begin].m_UserData;
+        const SpineModelComponent* first = (const SpineModelComponent*) world->m_Components.m_Objects[component_index];
         const SpineModelResource* resource = first->m_Resource;
 
         uint32_t vertex_start = world->m_VertexBufferData.Size();
         uint32_t vertex_count = 0;
         for (uint32_t *i = begin; i != end; ++i)
         {
-            const SpineModelComponent* component = (SpineModelComponent*) buf[*i].m_UserData;
+            uint32_t comp_index = (uint32_t)buf[*i].m_UserData;
```

Comment:
This second variable is very close to the `component_index`.
Please reuse it as it's currently confusing.

Reply:
I tried a different name because `component_index` was indexing the first array element and worried not to spoil anything. Under a closer look i see that it's not used again until the end of the function. 

So, yes, i'll re-use it.